stan/math/rev/core/chainablestack_inst.cpp

```diff
@@ -0,0 +1,14 @@
+#ifndef STAN_MATH_REV_CORE_CHAINABLESTACK_INST_CPP
+#define STAN_MATH_REV_CORE_CHAINABLESTACK_INST_CPP
+
+#include <stan/math/rev/core/chainablestack.hpp>
+
+namespace stan {
+namespace math {
+namespace internal {
+
+const ChainableStack::AutodiffStackStorage* __chainable_stack = init();
```

Comment:
I guess we can't just call init() at the top-level, right?

Would you mind clarifying why doing this here doesn't cause the TLS guards to be put back in?

Reply:
Oh! I see what's going on. I would change the name to something like `__unused_adstack` and put a comment in there explaining that the assignment is just to trick the compiler to trigger the init function. Empty namespace is not as useful as it would seem when you have a header-only library and everything in one TU - maybe use `internal` or make a new one for total trash objects like this, haha.